### PR TITLE
add `stZeroKnowledge` and `stZeroKnowledge2` testsuites

### PR DESCRIPTION
**Pull request by TAdev0**
Time spent on this PR: 60min

#438

# Pull Request type

Please check the type of change your PR introduces:

-   [ ] Bugfix
-   [ ] Feature
-   [ ] Code style update (formatting, renaming)
-   [ ] Refactoring (no functional changes, no API changes)
-   [ ] Build-related changes
-   [ ] Documentation content changes
-   [x] Testing

# What is the new behavior?

# Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: crates/ef-testing/tests/tests.rs [hunk 1]
```diff
@@ -127,6 +127,9 @@ mod blockchain_tests {
     // blockchain_tests!(st_transition_test, stTransitionTest); // 3 failing tests
     blockchain_tests!(st_wallet_test, stWalletTest);
     blockchain_tests!(st_zero_calls_revert, stZeroCallsRevert);
+    // blockchain_tests!(st_zero_calls_test, stZeroCallsTest); // 1 failing test
+    blockchain_tests!(st_zero_knowledge, stZeroKnowledge);
+    blockchain_tests!(st_zero_knowledge2, stZeroKnowledge2);
     blockchain_tests!(st_zero_calls_test, stZeroCallsTest);
     // blockchain_tests!(st_zero_knowledge, stZeroKnowledge); // 60 failing
```
Comment (danilowhk) [inline crates/ef-testing/tests/tests.rs:134]: These tests have been duplicated

Comment (TAdev0) [reply crates/ef-testing/tests/tests.rs:141]: stZeroKnowledge and stZeroKnowledge2 are 2 different tests, what do you mean?

Comment (danilowhk) [reply crates/ef-testing/tests/tests.rs:141]: Here you have:

    blockchain_tests!(st_zero_knowledge, stZeroKnowledge);

and 

    // blockchain_tests!(st_zero_knowledge, stZeroKnowledge); // 60 failing

If we implement a test we should just remove the comment, not add another line

**Comment by TAdev0** [comment]
@danilowhk  now it should be good :)

**Review by danilowhk — DISMISSED** [review]

**Comment by danilowhk** [comment]
Can merge once rebased

File: crates/ef-testing/tests/tests.rs [hunk 2]
```diff
@@ -127,8 +127,8 @@ mod blockchain_tests {
     blockchain_tests!(st_transition_test, stTransitionTest);
     blockchain_tests!(st_wallet_test, stWalletTest);
     blockchain_tests!(st_zero_calls_revert, stZeroCallsRevert);
+    blockchain_tests!(st_zero_knowledge, stZeroKnowledge);
+    blockchain_tests!(st_zero_knowledge2, stZeroKnowledge2);
```
Comment (danilowhk) [inline crates/ef-testing/tests/tests.rs:131]: Could you move the order of these tests to be in the same order as GeneralTests folder?

**Review by danilowhk — APPROVED** [review]

**Pull request merged**